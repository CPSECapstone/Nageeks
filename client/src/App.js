import React, {useState, setState} from 'react';
import Applistings from './components/Applisting';
import Customers from './components/customers';
import EventCalendar from './components/eventCalendar';
import Sidebar from './components/Sidebar';
import Banner from './components/Banner';
import Form from './components/form';
import Login from './Login';
import { Grid, Paper } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles({
  sidebarStyle: {
    backgroundColor: "#323236"
  },
  mainContentStyle: {
    color: "whitesmoke",
    backgroundColor: "#444449",
    fontSize: "18px",
    justify: "center",
  },
})

function App() {
  const [pageNum, setPageNum] = useState(0)
  const classes = useStyles();
  
  function displayPage() {
    let page;
    if (pageNum === 0) {
      page = <Applistings/>
    } else if (pageNum === 1) {
      page = <Customers />
    } else if (pageNum === 2) {
      page = <EventCalendar />
    }else if (pageNum === 3){
      page = <Login/>
    }
    return page;
  }

  return (
    <Grid 
      container
      direction="column"
    >
      <Grid 
        item 
        xs={12}
        style={{minHeight: '50px', maxHeight: '6vh'}}
      >
        <Banner/>
      </Grid>
      <Grid item xs={12}>
        <Grid 
          container
          direction="row"
          style={{ minHeight: '94vh' }}
        >
          <Grid 
            item xs={2}
            className={classes.sidebarStyle}
          >
            <Sidebar cb={setPageNum}/>
          </Grid>
          <Grid 
            item xs={10}
            className={classes.mainContentStyle}
          >
            {displayPage()}
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
}

export default App;

