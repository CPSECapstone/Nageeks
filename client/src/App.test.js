import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';

afterEach(cleanup);
test("First snapshot test", ()=>{
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
})
//tutorial for React Testing
//https://www.youtube.com/watch?v=3e1GHCA3GP0