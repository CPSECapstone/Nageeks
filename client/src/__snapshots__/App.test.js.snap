// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
<div
  className="App"
>
  <div
    className="banner"
  >
    <text
      className="banner-text"
    >
      CloudHaven
    </text>
    <text
      className="banner-account"
    >
       Welcome, Johnny
    </text>
  </div>
  <div
    className="home"
    pageNum={0}
  >
    <div
      className="side-bar"
    >
      <button
        className="side-bar-active"
        onClick={[Function]}
      >
        Apps
      </button>
      <button
        className="side-bar-button"
        onClick={[Function]}
      >
        My Data
      </button>
      <button
        className="side-bar-button"
        onClick={[Function]}
      >
        Calendar
      </button>
      <div
        className="sidebar-fill"
      />
    </div>
    <header
      className="App-header"
    >
      <div
        className="applist"
      >
        <div
          className="applisting-box"
        >
          <img
            alt="UC Davis Medical"
            className="app-pic"
            height="130"
            src="https://npuh82iut7x3aosxba3ol14m-wpengine.netdna-ssl.com/wp-content/uploads/2018/11/ucdavismedical-740x675.jpg"
            width="130"
          />
          <p
            className="app-title-text"
          >
             UC Davis Medical Portal
          </p>
        </div>
        <div
          className="applisting-box"
        />
        <div
          className="applisting-box"
        />
        <div
          className="applisting-box"
        />
      </div>
    </header>
  </div>
</div>
`;
